src/zero_copy/primitive.rs

```diff
@@ -511,3 +511,81 @@ macro_rules! impl_choice_for_tuple {
 }
 
 impl_choice_for_tuple!(A_ B_ C_ D_ E_ F_ G_ H_ I_ J_ K_ L_ M_ N_ O_ P_ Q_ S_ T_ U_ V_ W_ X_ Y_ Z_);
+
+#[derive(Copy, Clone)]
+pub struct Group<T> {
+    parsers: T,
+}
+
+pub const fn group<T>(parsers: T) -> Group<T> {
+    Group { parsers }
+}
+
+// recursively combine to flatten a tuple
+macro_rules! recursive_combine {
```

Comment:
Could this be simplified to a recursive `M::map`? I think I already tried that, but I'm not sure.

Reply:
Do you have a usage example of this?

Reply:
I've added a test to try and demonstrate it